### PR TITLE
Feature/mock refactor

### PR DESCRIPTION
**Pull request by Jammy2211**
Clean up of the mock package, inculding:

- Removal of mock_search and merging of the separate `MockSearch` / `MockResult` and other objects they had.
- Removal of `mock_real` in favour of abstract model objects.
- Clean up of tests.

### CODE REVIEW

File: autofit/config/doc [hunk 1]
```diff
@@ -11,7 +11,7 @@ priors:
         The default priors used on every model component in PyAutoFit (e.g. light proiles, mass profiles, etc.), for
         example if they use a UniformPrior or GausssianPrior and the range of values sampled.
     limit:
-        The limits between which a parameter may be sampled. For example, the axis_ratio of light and mass profiles has
+        The limits between which a parameter may be sampled. For example, the one of light and mass profiles has
```
Comment (rhayes777) [inline autofit/config/doc:14]: Did you mean to make this change?

File: autofit/example/model.py [hunk 2]
```diff
@@ -50,3 +51,45 @@ def profile_from_xvalues(self, xvalues):
...
+            np.exp(-0.5 * np.square(np.divide(transformed_xvalues, self.sigma))),
+        )
+
+    def inverse(
+            self,
+            y
+    ):
+        print(y)
+        print(f"centre = {self.centre}")
+        print(f"sigma = {self.sigma}")
```
Comment (rhayes777) [inline autofit/example/model.py:83]: Should these print statements be here?

File: test_autofit/database/identifier/test_identifiers.py [hunk 3]
```diff
@@ -329,14 +329,14 @@ def test__identifier_description():
     assert description[i] == "gaussian"; i+=1
     assert description[i] == "PriorModel"; i+=1
     assert description[i] == "cls"; i+=1
-    assert description[i] == "autofit.mock.mock.Gaussian"; i+=1
+    assert description[i] == "autofit.example.model.Gaussian"; i+=1
```
Comment (rhayes777) [inline test_autofit/database/identifier/test_identifiers.py:332]: Maybe use enumerate to make this test a bit more concise? dunno

File: test_autofit/graphical/info/test_info.py [hunk 4]
```diff
@@ -37,23 +37,23 @@ def test_non_trivial_info(
     assert info == """PriorFactors
 
 PriorFactor0 (AnalysisFactor1.sigma)                                                      UniformPrior, lower_limit = 0.0, upper_limit = 1.0
-PriorFactor1 (AnalysisFactor1.intensity)                                                  UniformPrior, lower_limit = 0.0, upper_limit = 1.0
+PriorFactor1 (AnalysisFactor1.normalization)                                              UniformPrior, lower_limit = 0.0, upper_limit = 1.0
```
Comment (rhayes777) [inline test_autofit/graphical/info/test_info.py:40]: These are the kinds of changes that affect identifiers

**Review by rhayes777 — APPROVED** [review]

File: autofit/example/model.py (hunk 2 above)
Comment (Jammy2211) [reply autofit/example/model.py:83]: Idk you put them there :P

File: test_autofit/graphical/info/test_info.py (hunk 4 above)
Comment (Jammy2211) [reply test_autofit/graphical/info/test_info.py:40]: This model component is only used for testing / documentation so its fine, but indeed I wouldnt be doing changes like this in autolens without a lot of care.

File: autofit/example/model.py (hunk 2 above)
Comment (rhayes777) [reply autofit/example/model.py:83]: Ah right worth removing them

**Pull request merged**